src/tiledb/cloud/geospatial/ingestion.py

```diff
@@ -0,0 +1,1287 @@
+            logger = get_logger_wrapper(verbose)
+
+            if append:
+                for f in sources:
+                    logger.debug("ingesting geometry dataset %r", f)
+                    with fiona.open(f) as colxn:
+                        with fiona.open(dataset_uri, mode="a") as dst:
+                            for feat in colxn:
+                                # OGR buffers before writing
+                                dst.write(feat)
+            else:
+                if extents and crs:
+                    logger.debug("creating geometry schema for %r", dataset_uri)
+                    if compressor:
+                        compressor_args = dict(compressor)
+                        compressor_name = compressor_args.pop("_name")
+                        if compressor_name:
+                            compressor_args = {
+                                k: None if not v else v
+                                for k, v in compressor_args.items()
+                            }
+                            compression_filter = vars(filter).get(compressor_name)(
+                                **compressor_args
+                            )
+                            compression_level = compression_filter.level
+
+                        else:
+                            raise ValueError
+
+                        with fiona.open(
+                            dataset_uri,
+                            driver="TileDB",
+                            schema=schema,
+                            crs=crs,
+                            BATCH_SIZE=chunk_size,
+                            TILE_CAPACITY=chunk_size,
+                            COMPRESSION=compressor_name[
+                                :-6
+                            ].upper(),  # remove trailing `Filter`
+                            COMPRESSION_LEVEL=compression_level,
+                            BOUNDS=",".join(str(v) for v in extents),
+                        ):
+                            pass
+                    else:
+                        with fiona.open(
+                            dataset_uri,
+                            mode="w",
+                            driver="TileDB",
+                            schema=schema,
+                            crs=crs,
+                            BATCH_SIZE=chunk_size,
+                            TILE_CAPACITY=chunk_size,
+                            BOUNDS=",".join(str(v) for v in extents),
+                        ) as dst:
+                            pass
+
+                else:
+                    raise ValueError("Insufficient metadata for point cloud ingestion")
+
+            logger.info("max memory usage: %.3f GiB", max_memory_usage() / (1 << 30))
+
+
+def ingest_point_cloud_udf(
+    sources: Optional[Sequence[str]] = None,
+    *,
+    dataset_uri: str,
+    template_sample: str,
+    append: bool = False,
+    extents: Optional[Tuple[float, float, float, float, float, float]] = None,
```

Comment:
This might be worth defining a type alias for, and/or breaking down:

```python
SixTuple = Tuple[float, float, float, float, float, float]
# or maybe:
XYZ = Tuple[float, float, float]
XYZPair = Tuple[XYZ, XYZ]

# so this becomes
def ingest_point_cloud_udf(
    ...,
    offsets: Optional[SixTuple] = None,
    # or maybe
    scales: Optional[XYZPair] = None,
```

(In both cases I'm sure you can come up with a better name.)

Reply:
👍